plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -0,0 +1,218 @@
+ */
+
+package org.opensearch.ml.processor;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+/**
+ * General ModelExecutor interface.
+ */
+public interface ModelExecutor {
+
+    Configuration suppressExceptionConfiguration = Configuration
+        .builder()
+        .options(Option.SUPPRESS_EXCEPTIONS, Option.DEFAULT_PATH_LEAF_TO_NULL)
+        .build();
+
+    default <T> ActionRequest getRemoteModelInferenceRequest(Map<String, String> parameters, String modelId) {
+        if (parameters == null) {
+            throw new IllegalArgumentException("wrong input. The model input cannot be empty.");
+        }
+        RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
+
+        MLInput mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build();
+
+        ActionRequest request = new MLPredictionTaskRequest(modelId, mlInput, null);
+
+        return request;
+
+    }
+
+    /**
+     * Read ModelTensorOutput and find value in DataAsMap and Data based on field Name
+     * @param modelTensorOutput
+     * @param originalModelOutputFieldName
+     * @return
+     */
+    default Object getModelOutputValue(ModelTensorOutput modelTensorOutput, String originalModelOutputFieldName, boolean ignoreMissing) {
+        Object modelOutputValue;
+        try {
+            // getMlModelOutputs() returns a list or collection.
+            // Adding null check for modelTensorOutput
+            if (modelTensorOutput != null && !modelTensorOutput.getMlModelOutputs().isEmpty()) {
+                // getMlModelOutputs() returns a list of ModelTensors
+                // accessing the first element.
+                ModelTensors output = modelTensorOutput.getMlModelOutputs().get(0);
+                // Adding null check for output
+                if (output != null && output.getMlModelTensors() != null && !output.getMlModelTensors().isEmpty()) {
+                    // getMlModelTensors() returns a list of ModelTensor
+                    if (output.getMlModelTensors().size() == 1) {
+                        ModelTensor tensor = output.getMlModelTensors().get(0);
```

Comment:
Suggest add some todo here to check how to support multiple tensors in future.

Reply:
added TODO

Reply:
https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java#L81